account_move_name_sequence/models/account_move.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2021 Akretion France (http://www.akretion.com/)
+# @author: Alexis de Lattre <alexis.delattre@akretion.com>
+# @author: Moisés López <moylop260@vauxoo.com>
+# @author: Francisco Luna <fluna@vauxoo.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    name = fields.Char(compute="_compute_name_by_sequence")
+    # highest_name, sequence_prefix and sequence_number are not needed any more
+    # -> compute=False to improve perf
+    highest_name = fields.Char(compute=False)
+    sequence_prefix = fields.Char(compute=False)
+    sequence_number = fields.Integer(compute=False)
+
+    _sql_constraints = [
+        (
+            "name_state_diagonal",
+            "CHECK(COALESCE(name, '') NOT IN ('/', '') OR state!='posted')",
+            "A move can not be posted with name '/' or empty value\n"
+            "Check the journal sequence, please",
+        ),
+    ]
+
+    @api.depends("state", "journal_id", "date")
+    def _compute_name_by_sequence(self):
+        for move in self:
+            name = move.name or "/"
+            # I can't use posted_before in this IF because
+            # posted_before is set to True in _post() at the same
+            # time as state is set to "posted"
+            if (
+                move.state == "posted"
+                and (not move.name or move.name == "/")
+                and move.journal_id
+                and move.journal_id.sequence_id
+            ):
+                if (
+                    move.move_type in ("out_refund", "in_refund")
+                    and move.journal_id.type in ("sale", "purchase")
+                    and move.journal_id.refund_sequence
+                    and move.journal_id.refund_sequence_id
+                ):
+                    seq = move.journal_id.refund_sequence_id
+                else:
+                    seq = move.journal_id.sequence_id
+                name = seq.next_by_id(sequence_date=move.date)
+            move.name = name
+
+    # We must by-pass this constraint of sequence.mixin
+    def _constrains_date_sequence(self):
+        return True
+
+    def _is_end_of_seq_chain(self):
```

Comment:
Why wasn't it required in v14?

Take into account migration should only include migration commits. If this is a fix, then it should be done in v14

Reply:
Odoo v14 is not using these methods

These methods are new only in v15.0

Reply:
Then this change should be included into the migration commit instead of an additional [FIX] commit, as it's part of the migration.